### PR TITLE
FormsUiHelper for UmbracoForms

### PR DESCRIPTION
**Pull request by dtng95**
- Add new FormsUiHelper which contains methods used in UmbracoForms acceptance tests for V14.

### CODE REVIEW

File: lib/helpers/FormsUiHelper.ts [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+  }
+
+  async ensureFormModerationSetting(){
+    await expect(this.formSettingModeration.locator('uui-label[title="manualApproval"]')).toBeVisible();
+    const enablePostModerationAttributeToggleInput = this.formSettingModeration.locator('uui-toggle #slider');
+    await expect(enablePostModerationAttributeToggleInput).toBeVisible();
+    await enablePostModerationAttributeToggleInput.click();
+  }
+
+  async ensureFormFieldsDisplayedSetting(){
```
Comment (andr317c) [inline lib/helpers/FormsUiHelper.ts:221]: Most of these ensure namings don't quite match what is actually happening in the methods.

**Review by andr317c — CHANGES_REQUESTED** [review]
Had some comments

File: lib/helpers/FormsUiHelper.ts (hunk 1 above)
Comment (AndyButland) [reply lib/helpers/FormsUiHelper.ts:221]: See my comment above for suggestion on naming.

File: lib/helpers/FormsUiHelper.ts [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+
+  async clickFormWorkflowConfigureSubmitButton() {
+    await this.formConfigureWorkflowModal.locator(this.formSubmitButtonModal).click();
+  }
+
+  async clickFormWorkflowAddButton(){
+    await this.formWorkflowOnSubmitStage.locator(this.formWorkflowAddButtonModal).click({force: true});
+  }
+
+  //TODO: Refactor this part. Would be good to have a `match` property for locator, in order to get the exact text.
```
[the remark this answers is not in the document]
Comment (dtng95) [reply lib/helpers/FormsUiHelper.ts:426]: let's add additional markup to Forms. It should be quick to do so once we merged that, I will update this part of FormsUIHelper.

**Review by AndyButland — APPROVED** [review]
Looks good from my perspective now, thanks.

File: lib/helpers/FormsUiHelper.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dtng95) [reply lib/helpers/FormsUiHelper.ts]: I created a PR for this refactoring: https://github.com/umbraco/Forms/pull/1105

File: lib/helpers/FormsUiHelper.ts [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+    await expect(cssClassInput).toBeVisible();
+    await cssClassInput.fill("custom-css-class");
+    const disableDefaultStylesheetInput = this.formSettingStyling.locator('uui-toggle #slider');
+    await expect(disableDefaultStylesheetInput).toBeVisible();
+    await disableDefaultStylesheetInput.click();
+  }
+
+  async ensureFormValidationSetting(){
+    await expect(this.formSettingValidation.locator('uui-label[title="validation"]')).toBeVisible();
+    const requiredErrorMessageInput = this.formSettingValidation.locator("input[type = 'text']").nth(0);
```
[the remark this answers is not in the document]
Comment (dtng95) [reply lib/helpers/FormsUiHelper.ts]: currently this is the only way I can find to handle this case that we need to test, and it's only used 1 time only so from my perspective it shouldn't be a problem

File: lib/helpers/FormsUiHelper.ts [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+  private readonly formSettingValidation : Locator;
+  private readonly formSettingAutocomplete : Locator;
+  private readonly formSettingModeration : Locator;
+  private readonly formSettingFieldsDisplayed : Locator;
+  private readonly formSettingDataRetention : Locator;
+  private readonly formSettingCaptionsLable : Locator;
+  private readonly formSettingStylingLable : Locator;
+  private readonly formSettingValidationLable : Locator;
+  private readonly formSettingAutocompleteLable : Locator;
+  private readonly formSettingModerationLable : Locator;
```
Comment (andr317c) [inline lib/helpers/FormsUiHelper.ts:38]: Should be label

File: lib/helpers/FormsUiHelper.ts [hunk 5]
```diff
@@ -0,0 +1,458 @@
...
+  private readonly formSettingFieldsDisplayedLabel : Locator;
+  private readonly formSettingDataRetentionLable : Locator;
```
Comment (andr317c) [inline lib/helpers/FormsUiHelper.ts:40]:
```suggestion
  private readonly formSettingDataRetentionLabel : Locator;
```

**Review by andr317c — CHANGES_REQUESTED** [review]
Had a few more comments

**Comment by dtng95** [comment]
Close this PR and create a new one to merge in v15/dev in order to avoid major conflicts

**Pull request closed without merging**